sdk/src/types/block/macro.rs

```diff
@@ -200,3 +200,38 @@ macro_rules! impl_serde_typed_dto {
         }
     };
 }
+
+#[macro_export]
+macro_rules! def_is_as_opt {
+    ($type:ty: $($name:ident),+$(,)?) => {
+        paste::paste! {
+            $(
+                #[doc = "Checks whether the " [<$type:snake>] " is a(n) [`" [<$name $type>] "`]."]
```

Comment:
```suggestion
                #[doc = "Checks whether the " [<$type:snake>] " is a(n) [`" [<$name>] "`]."]
```
Since $name always ends with $type, I wouldn't put type